server/src/main/java/com/unosquare/carmigo/controller/AdminUserController.java

```diff
@@ -0,0 +1,92 @@
+package com.unosquare.carmigo.controller;
+
+import com.github.fge.jsonpatch.JsonPatch;
+import com.unosquare.carmigo.model.request.CreateDriverViewModel;
+import com.unosquare.carmigo.model.response.DriverViewModel;
+import com.unosquare.carmigo.model.response.PassengerViewModel;
+import com.unosquare.carmigo.model.response.PlatformUserViewModel;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import javax.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/v1/admin/users")
+@Tag(name = "Admin User Controller")
+public class AdminUserController {
+
+  private final UserControllerHelper userControllerHelper;
+
+  @GetMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+  @ResponseStatus(HttpStatus.OK)
+  public ResponseEntity<PlatformUserViewModel> getPlatformUserById(@PathVariable final int id) {
+    return ResponseEntity.ok(userControllerHelper.getPlatformUserById(id));
+  }
+
+  @PatchMapping(value = "/{id}", consumes = "application/json-patch+json")
+  @ResponseStatus(HttpStatus.ACCEPTED)
+  public ResponseEntity<PlatformUserViewModel> patchPlatformUserById(
+      @PathVariable final int id, @RequestBody final JsonPatch patch) {
+    return new ResponseEntity<>(userControllerHelper.patchPlatformUserById(id, patch), HttpStatus.ACCEPTED);
+  }
+
+  @DeleteMapping(value = "/{id}")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  public ResponseEntity<?> deletePlatformUserById(@PathVariable final int id) {
+    userControllerHelper.deletePlatformUserById(id);
```

Comment:
Consider the thought that the action of deleting a user would be carried out by a userService in line with the typical Controller>Service>Repo pattern.  A helper of a controller does not seem like the right place.

Reply:
The controller helper does not know about repositories so it does not delete a user. It acts as a man-in-the-middle by receiving calls from UserController and AdminUserController and forwarding them to UserService.

Reply:
UserControllerHelper removed.